subsys/bluetooth/mesh/sensor_types.c

```diff
@@ -471,6 +514,244 @@ static int float32_decode(const struct bt_mesh_sensor_format *format,
 	val->val2 = ((int64_t)(fvalue * 1000000.0f)) % 1000000L;
 	return 0;
 }
+#else
+static int scalar_compare(const struct bt_mesh_sensor_value *op1,
+			  const struct bt_mesh_sensor_value *op2)
+{
+	int64_t op1_raw = scalar_decode_raw(op1->format, op1->raw);
+	int64_t op2_raw = scalar_decode_raw(op2->format, op2->raw);
+
+	if (op1_raw > op2_raw) {
+		return 1;
+	} else if (op1_raw < op2_raw) {
+		return -1;
+	} else {
+		return 0;
+	}
+}
+
+static bool percentage_delta_check(const struct bt_mesh_sensor_value *delta,
+				   float diff, float prev)
+{
+	if (prev == 0.0f) {
+		return true;
+	}
+
+	float percentage_float = scalar_decode_raw(delta->format, delta->raw) / 10000.0f;
```

Comment:
No, for percentage delta there is always a scalar stored in `delta->raw`. Not matter the format of the sensor channel in question, the percentage is always a scalar. I cast the scalar value to float because float-math was the best common denominator I could find for comparing ([any format] - [any format] > [scalar])